packages/components/src/components/hds/icon/index.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { guidFor } from '@ember/object/internals';
+import { assert } from '@ember/debug';
+import { iconNames } from '@hashicorp/flight-icons/svg';
+
+import type { IconName } from '@hashicorp/flight-icons/svg';
+
+export interface HdsIconSignature {
+  Args: {
+    name?: IconName;
+    color?: string;
+    size?: '16' | '24';
```

Comment:
question: should we extract this as an enum to be able to use it in other components? or do we prefer to reference it from here (e.g. `HdsIconSignature['Args']['size]`)?

Reply:
What would we call the enum values? Is there another terminology for these sizes aside from a numerical size?

Reply:
see these examples:
- https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/separator/types.ts#L7-L8
- https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/text/types.ts#L48-L52